### ISSUE TITLE
Misleading diagnostic from ambiguous reference to class template

### ISSUE THREAD

**Issue by rjmccall** [llvm/llvm-project#13765]
|  |  |
| --- | --- |
| Bugzilla Link | [13393](https://llvm.org/bz13393) |
| Version | trunk |
| OS | All |
| CC | @DougGregor,@zygoloid |

## Extended Description 
jeweloflife:clang$ cat /tmp/red.cpp
```c++
template <class T> class A;
namespace inner {
  template <class T> class A;
}
using namespace inner;

class B : A<B> {};
```

jeweloflife:clang$ clang /tmp/red.cpp
```
/tmp/red.cpp:7:11: error: unknown template name 'A'
class B : A<B> {};
          ^
1 error generated.
```

jeweloflife:clang$ cat << EOF > /dev/null

Obviously this diagnostic should say something about an *ambiguous* template reference, not an *unknown* one.

**Comment by philnik777** [comment]
Fixed in Clang 9: https://godbolt.org/z/M181KjTra
